src/prefect/events/schemas/automations.py

```diff
@@ -0,0 +1,311 @@
+import abc
+from datetime import timedelta
+from enum import Enum
+from typing import (
+    Any,
+    Dict,
+    List,
+    Literal,
+    Optional,
+    Set,
+    Union,
+)
+from uuid import UUID
+
+from typing_extensions import TypeAlias
+
+from prefect._internal.pydantic import HAS_PYDANTIC_V2
+
+if HAS_PYDANTIC_V2:
+    from pydantic.v1 import Extra, Field, root_validator, validator
+    from pydantic.v1.fields import ModelField
+else:
+    from pydantic import Extra, Field, root_validator, validator
+    from pydantic.fields import ModelField
+
+from prefect._internal.schemas.bases import PrefectBaseModel
+from prefect.events.actions import ActionTypes
+from prefect.utilities.collections import AutoEnum
+
+from .events import ResourceSpecification
+
+
+class Posture(AutoEnum):
+    Reactive = "Reactive"
+    Proactive = "Proactive"
+    Metric = "Metric"
+
+
+class Trigger(PrefectBaseModel, abc.ABC):
+    """
+    Base class describing a set of criteria that must be satisfied in order to trigger
+    an automation.
+    """
+
+    class Config:
```

Comment:
would you be willing to something (kinda gross, I know) for the sake of pydantic compat?

```python
In [1]: from pydantic import BaseModel

In [2]: class Foo(BaseModel, extra="allow"):
   ...:     x: int
   ...:

In [3]: Foo(x=42, y="bar")
Out[3]: Foo(x=42, y='bar')

In [4]: from pydantic.v1 import BaseModel

In [5]: class FooV1(BaseModel, extra="allow"):
   ...:     x: int
   ...:

In [6]: FooV1(x=42, y="bar")
Out[6]: FooV1(x=42, y='bar')

In [7]: !uv pip list | rg pyd
pydantic             2.6.4
```

Reply:
Oh no worries at all, can def do this